lib/aggregate/aggregate_store.rb

```diff
@@ -71,15 +73,45 @@ def self.included(model_class)
       model_class.extend ClassMethods
     end
 
+    def save
+      defined?(super) &&
+        RailsUpgradeHelpers::Version.if_version(
+          rails_5: -> { set_saved_changes },
+          rails_6: -> { set_saved_changes }
+        )
+      super
+    end
+
+    def save!
+      defined?(super) &&
+        RailsUpgradeHelpers::Version.if_version(
+          rails_5: -> { set_saved_changes },
+          rails_6: -> { set_saved_changes }
+        )
+      super
+    end
+
     def changed?
       (defined?(super) && super) || @changed
     end
 
+    def saved_changes?
+      RailsUpgradeHelpers::Version.if_version(
+        rails_4: -> { raise NoMethodError, "undefined method 'saved_changes?' for #{self}" }
+      )
+      (defined?(super) && super) || @saved_changes
+    end
+
     def set_changed
       @changed = aggregate_values != aggregate_initial_values
       aggregate_owner&.set_changed
     end
 
+    def set_saved_changes
+      @saved_changes = @changed
```

Comment:
so, this is potentially going to be confusing, since in Rails 5, `saved_changes` is also a method that returns **what the actual saved changes are**.

Do we need to track those as well?

Additionally, we need to handle things like `reload` that reset `@saved_changes`. We have a [helper method `reset_cache_data`](https://github.com/Invoca/aggregate/blob/33d1a80636d9129bbcca247cea0eb3a4ca6bd961/lib/aggregate/container.rb#L102) to facilitate that.